### PR TITLE
Improve language in docs

### PR DESCRIPTION
**Pull request by JoosepAlviste**
Hey! I recently started using this plugin and am very happy with it! Since I was reading through the documentation anyways, I decided to make some small language-specific changes/improvements there.

I'm not a native english speaker so there might still be some problems with my changes but I think it's a bit easier to read the docs now. Let me know if I should undo any changes or if you have any other ideas.

### CODE REVIEW

File: doc/coc.txt [hunk 1]
```diff
@@ -45,63 +45,67 @@ Changelog					|coc-changelog|
...
 
-Use command |:CocConfig| to open user configuration file, or create
-`.vim/coc-settings.json` in your project root for folder configuration.
+Use the command |:CocConfig| to open the user configuration file, or create
+`.vim/coc-settings.json` in your project root for folder-based configuration.
 
-Auto completion of json file would be enabled after `coc-json` extension
-installed.
+Auto completion in the settings file would be enabled after the `coc-json` 
+extension is installed.
```
Comment (chemzqm) [inline doc/coc.txt]: Should remove `the`.

File: doc/coc.txt [hunk 2]
```diff
@@ -45,63 +45,67 @@ Changelog					|coc-changelog|
...
 Completion is triggered automatically by default, you can change completion
-behavior by use configuration file.
+behavior in the configuration file.
 
 To disable automatic trigger, use: >
 
 	"suggest.autoTrigger": "none",
 <
-To enable trigger after insert enter, use: >
+To enable trigger after inserting enter, use: >
```
Comment (chemzqm) [inline doc/coc.txt]: Should be `insert enter`

File: doc/coc.txt [hunk 3]
```diff
@@ -118,38 +122,38 @@ To prevent overriding `completeopt`, use:
...
+To trigger completion with two input characters, use: >
 
 	"suggest.minTriggerInputLength": 2
 <
 To enable commit characters feature, use: >
 
 	"suggest.acceptSuggestionOnCommitCharacter": true
 <
-To change the indicator of snippet item, use: >
+To change the indicator of a snippet item, use: >
```
Comment (chemzqm) [inline doc/coc.txt]: Should be `snippet items`.

File: doc/coc.txt [hunk 4]
```diff
@@ -300,38 +305,38 @@ g:coc_user_config 				*g:coc_user_config*
...
-			define this variable to list of extension names when
-			you can't use |coc#add_extension()|
+			Global extension names to install when they aren't 
+			installed, define this variable to a list of extension 
+			names when you can't use |coc#add_extension()|
 
 g:coc_enable_locationlist 			*g:coc_enable_locationlist*
 
-			Open location list when locations changes.
+			Open location list when location changes.
```
Comment (chemzqm) [inline doc/coc.txt]: `location` => `location list`

File: doc/coc.txt [hunk 5]
```diff
@@ -300,38 +305,38 @@ g:coc_user_config 				*g:coc_user_config*
...
 
 g:coc_snippet_next                            	*g:coc_snippet_next*
 
-			Trigger key for goto next snippet position, applied
-			on insert and select mode.
+			Trigger key for going to the next snippet position, 
+			applied in insert and select mode.
 
-			Only works when snippet session is activated.
+			Only works when a snippet session is activated.
```
Comment (chemzqm) [inline doc/coc.txt]: remove `a`

File: doc/coc.txt [hunk 6]
```diff
@@ -300,38 +305,38 @@ g:coc_user_config 				*g:coc_user_config*
...
 
 g:coc_snippet_prev                            	*g:coc_snippet_prev*
 
-			Trigger key for goto previous snippet position,
-			applied on insert and select mode.
+			Trigger key for going to the previous snippet 
+			position, applied in insert and select mode.
 
-			Only works when snippet session is activated.
+			Only works when a snippet session is activated.
```
Comment (chemzqm) [inline doc/coc.txt]: remove `a`

File: doc/coc.txt (hunk 1 above)
Comment (Avi-D-coder) [reply doc/coc.txt]: One of these would sound more natural from my point of view.
"Auto completion in the settings file will be enabled after the `coc-json` extension is installed."
"To enable auto completion in the settings file, install the `coc-json` extension."

Comment (Avi-D-coder) [reply doc/coc.txt]: The "the" does change the semantics, but is technically correct.

**Pull request merged**